### ISSUE TITLE
Circular inclusion in TrackingTools/GsfTools

### ISSUE THREAD

**Issue by mrodozov** [cms-sw/cmssw#23633]
The last remaining header from the fsyntax check that we couldn't fix is this one:
TrackingTools/GsfTools/interface/MultiGaussianStateCombiner.h 
with this short description:
```
In file included from /build/cmsbld/jenkins/workspace/ib-run-check-headers/CMSSW_10_2_X_2018-06-20-2300/src/TrackingTools/GsfTools/interface/MultiGaussianState.h:112:0,
                 from src/TrackingTools/GsfTools/interface/MultiGaussianStateCombiner.h:4:
/build/cmsbld/jenkins/workspace/ib-run-check-headers/CMSSW_10_2_X_2018-06-20-2300/src/TrackingTools/GsfTools/interface/MultiGaussianState.icc: In member function 'void MultiGaussianState<N>::checkCombinedState() const':
/build/cmsbld/jenkins/workspace/ib-run-check-headers/CMSSW_10_2_X_2018-06-20-2300/src/TrackingTools/GsfTools/interface/MultiGaussianState.icc:52:3: error: 'MultiGaussianStateCombiner' was not declared in this scope
   MultiGaussianStateCombiner<N> theCombiner;
   ^~~~~~~~~~~~~~~~~~~~~~~~~~
/build/cmsbld/jenkins/workspace/ib-run-check-headers/CMSSW_10_2_X_2018-06-20-2300/src/TrackingTools/GsfTools/interface/MultiGaussianState.icc:52:33: error: 'theCombiner' was not declared in this scope
   MultiGaussianStateCombiner<N> theCombiner;
                                 ^~~~~~~~~~~
/build/cmsbld/jenkins/workspace/ib-run-check-headers/CMSSW_10_2_X_2018-06-20-2300/src/TrackingTools/GsfTools/interface/MultiGaussianState.icc: In member function 'void MultiGaussianState<N>::setWeight(double)':
/build/cmsbld/jenkins/workspace/ib-run-check-headers/CMSSW_10_2_X_2018-06-20-2300/src/TrackingTools/GsfTools/interface/MultiGaussianState.icc:62:12: error: 'DBL_MIN' was not declared in this scope
   if ( wgt>DBL_MIN )  rescaleWeight(newWeight/wgt);
            ^~~~~~~

```
Can we ask you to help us with that, as it requires expert with better understanding than us of this package, to avoid the circular inclusions present. 
To reproduce the issue
```
git-cms-addpkg TrackingTools/GsfTools
scram p -j 16 -k check-headers
```
References to the error:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/check_headers/CMSSW_10_2_X_2018-06-20-2300/slc6_amd64_gcc630/build.log

Thanks in advance,
Cheers,
M.

**Comment by VinInn** [comment]
My only solution would be to combine the two headers in one.

**Comment by perrotta** [comment]
As far as I can see, this got fixed by #26020 : @Dr15Jones, can you confirm (and close this issue, if so)?

**Comment by Dr15Jones** [comment]
@perrotta I thought we were just waiting for a reconstruction L2 to do sign off on this issue, then it will be closed.

**Comment by Dr15Jones** [comment]
@perrotta it does look like the problem was fixed in the referenced pull request.

**Comment by perrotta** [comment]
+1
- Solved with #26020: thank you @Dr15Jones for the confirmation
